core/site_battery.go

```diff
@@ -5,45 +5,43 @@ import (
 	"github.com/evcc-io/evcc/core/loadpoint"
 )
 
-// getBatteryMode returns the battery mode
-func (site *Site) getBatteryMode() api.BatteryMode {
+// GetBatteryMode returns the battery mode
+func (site *Site) GetBatteryMode() api.BatteryMode {
 	site.Lock()
 	defer site.Unlock()
 	return site.batteryMode
 }
 
-// setBatteryMode sets the battery mode
-func (site *Site) setBatteryMode(batMode api.BatteryMode) {
+// SetBatteryMode sets the battery mode
+func (site *Site) SetBatteryMode(batMode api.BatteryMode) {
 	site.Lock()
 	defer site.Unlock()
 	site.batteryMode = batMode
-	site.publish("batteryMode", batMode.String())
+	site.publish("batteryMode", batMode)
 }
 
-func (site *Site) updateBatteryMode(loadpoints []loadpoint.API) {
-	// determine expected state
-	batMode := api.BatteryNormal
+func (site *Site) loadpointBatteryMode(loadpoints []loadpoint.API) api.BatteryMode {
 	for _, lp := range loadpoints {
 		if lp.GetStatus() == api.StatusC && (lp.GetMode() == api.ModeNow || lp.GetPlanActive()) {
-			batMode = api.BatteryHold
-			break
+			return api.BatteryHold
 		}
 	}
 
-	if batMode == site.getBatteryMode() {
```

Comment:
@andig 
Removal of this, is what makes the test `TestBatteryModeNoUpdate` fail.
If it is intended to always update the battery mode on every interval, the test can simply be removed. Otherwise, this check needs to be added back into `updateBatteryMode(mode)` below.

Is this change IS intended, my follow up question would be if we expect battery controller implementations to cache the current mode and only send commands down the wire if needed.

Reply:
Das ist auf 2 Funktionen aufgeteilt.

Reply:
Ja, hatte ich gesehen. Was nicht ganz offensichtlich war - der check auf den aktuellen Bat mode ist jetzt in die core Logik gewandert. Der o.g. Test kann dann trotzdem raus, da wir nicht site.Update() testen.

Reply:
Ist erledigt: #10859 - merge please

Reply:
Tausend Dank!